src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -478,13 +478,14 @@ impl<'a> Linker for GccLinker<'a> {
     }
 
     fn debuginfo(&mut self, strip: Strip) {
+        // MacOS linker does not support longhand arguments --strip-debug and --strip-all
         match strip {
             Strip::None => {}
             Strip::Debuginfo => {
-                self.linker_arg("--strip-debug");
+                self.linker_arg("-s");
```

Comment:
```suggestion
                self.linker_arg("-S"); // --strip-debug
```
(It would be better to leave comments with full options.)

Reply:
Ah, ok, the updated version is good as is.